### DISCUSSION TITLE
v2 - Allow modification of rounding mode when calling `transformScale`

### DISCUSSION THREAD

**Discussion by tranhl** [dinerojs/dinero.js#647] [Ideas]
Related to #513 

My team and I are currently integrating with a payments processor that requires minor units in the request. The documentation for v2 suggests using `toSnapshot` to grab the amount in minor units, but this doesn't work when the scale for the Dinero object is greater than the currency's exponent.

This is where `transformScale` comes in (or so we thought). Right now, `transformScale` always truncates precision when going from a larger scale to a smaller scale.

I suggest allowing us to provide a rounding mode (e.g. `halfUp`) when calling `transformScale` to allow flexibility in how we manage the loss of precision.

For example, this is how this new feature could look like in code:

```ts
const d = dinero({ amount: 12345, currency: USD, scale: 3 })
const scaled = transformScale(d, { newScale: d.toJSON().currency.exponent, round: halfUp })

console.log(scaled.toJSON()) // { amount: 1235, scale: 2, currency: ... }
```

Note how the scaled Dinero's amount is `1235` instead of `1234`, which is what we would get with current behavior.

**Comment by sarahdayan** [comment]
Hey @tranhl, I think this would be awesome to provide a way to handle how `transformScale` handles lossy transformations.

**Do you want to take a stab at it?**

If you do, I'd recommend branching off of https://github.com/dinerojs/dinero.js/pull/309 which changes a bunch of things in a breaking way. I'm expecting to merge it soon, so this would be safer.

**Reply by sarahdayan** [reply]
I had forgotten that #309 actually handles it by letting you pass a custom `divide` function (see [preview docs](https://dinerojs-git-feat-to-units-dinerojs.vercel.app/docs/api/conversions/transform-scale#pass-a-custom-divide-function)).
